### PR TITLE
implement new compiler checks for NVHPC

### LINKED ISSUE #2815: Implement new way of compiler (+ version) checks for NVHPC
Implement new way of compiler (+ version) checks for NVHPC as was discussed and decided [here](https://github.com/NVIDIA/cccl/issues/2784#issuecomment-2476398592). Same approach should be applied to all other compilers in the future.

### PR DESCRIPTION
**Pull request by davebayer**
## Description

Closes #2815.

Implement new compiler checks for NVHPC only.

## Checklist
- [x] I am familiar with the [Contributing Guidelines]().
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: libcudacxx/include/cuda/std/__cccl/compiler.h [hunk 1]
```diff
@@ -19,7 +21,7 @@
       "Support for the Intel C++ Compiler Classic is deprecated and will eventually be removed. Define CCCL_SUPPRESS_ICC_DEPRECATION_WARNING to suppress this warning"
 #  endif // CCCL_SUPPRESS_ICC_DEPRECATION_WARNING
 #elif defined(__NVCOMPILER)
-#  define _CCCL_COMPILER_NVHPC
+#  define _CCCL_COMPILER_NVHPC _CCCL_COMPILER_MAKE_VERSION(__NVCOMPILER_MAJOR__, __NVCOMPILER_MINOR__)
 #  define _CCCL_COMPILER_NVHPC_VERSION \
     (__NVCOMPILER_MAJOR__ * 10000 + __NVCOMPILER_MINOR__ * 100 + __NVCOMPILER_PATCHLEVEL__)
```
Comment (miscco) [inline libcudacxx/include/cuda/std/__cccl/compiler.h:26]: We should drop that in favor of just `_CCCL_COMPILER_NVHPC`
```suggestion
```

**Review by miscco — APPROVED** [review]

File: libcudacxx/include/cuda/std/__cccl/compiler.h (hunk 1 above)
Comment (davebayer) [reply libcudacxx/include/cuda/std/__cccl/compiler.h]: I wanted to keep all compiler version macros unchanged for now as Thrust compiler marcos depend on them and remove them in separate PR. But the fact is that nvhpc version macros is unused, so there is no problem removing it now :)

**Review by ericniebler — APPROVED** [review]
this is a good direction i think

**Comment by miscco** [comment]
Thanks a lot for improving our core infrastructure 🎉

**Pull request merged**